omisego-sdk/src/main/java/co/omisego/omisego/custom/Callback.kt

```diff
@@ -21,12 +21,12 @@ interface Callback<in T> {
      *
      * @param response The serialization of the provided associated data
      */
-    fun success(response: OMGResponse<T>)
+    fun success(response: T)
```

Comment:
I've already tried that before I submit. I can't do so I had forced to change it from `OMGResponse<T>` to `T`

The reason is when I use `OMGResponse<T>` in the `Callback<T>` like this

```kotlin
interface Callback<in T> {
    fun success(response: OMGResponse<T>)
    fun fail(response: OMGResponse<APIError>)
}
```
Also in the interface `EWalletAPI`, we should also remove the `OMGResponse` (or we will get OMGResponse<OMGResponse<User>> in the `success` callback which is wrong) so it should be look like this.

```kotlin
interface EWalletAPI {
    @POST(GET_CURRENT_USER)
    fun getCurrentUser(): OMGCall<User>

    @POST(LOGOUT)
    fun logout(): OMGCall<Logout>

    @POST(LIST_BALANCE)
    fun listBalances(): OMGCall<BalanceList>

    @POST(GET_SETTINGS)
    fun getSettings(): OMGCall<Setting>
}
```

Now the problem arises because the response is coming in the following format.

```
{
  "version": "1",
  "success": true,
  "data": {
    "username": "user01",
    "provider_user_id": "provider_user_id01",
    "object": "user",
    "metadata": {},
    "id": "48236187-9c5c-4568-8cdf-f0233d035574"
  }
}
```

and boom! something wrong we can't parse the response now. Maybe I need to fix the converter 🤔